domain/domain_test.go

```diff
@@ -40,88 +39,43 @@ import (
 	"github.com/pingcap/tidb/store/mockstore"
 	"github.com/pingcap/tidb/util"
 	"github.com/pingcap/tidb/util/mock"
-	"github.com/pingcap/tidb/util/testleak"
 	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/tikv/client-go/v2/oracle"
 	"github.com/tikv/client-go/v2/tikv"
 	"go.etcd.io/etcd/integration"
 )
 
-func TestT(t *testing.T) {
-	CustomVerboseFlag = true
-	TestingT(t)
-}
-
-var _ = Suite(&testSuite{})
-
-type testSuite struct {
-}
-
-func mockFactory() (pools.Resource, error) {
-	return nil, errors.New("mock factory should not be called")
-}
-
-func sysMockFactory(dom *Domain) (pools.Resource, error) {
-	return nil, nil
-}
-
-type mockEtcdBackend struct {
-	kv.Storage
-	pdAddrs []string
-}
-
-func (mebd *mockEtcdBackend) EtcdAddrs() ([]string, error) {
-	return mebd.pdAddrs, nil
-}
-func (mebd *mockEtcdBackend) TLSConfig() *tls.Config { return nil }
-func (mebd *mockEtcdBackend) StartGCWorker() error {
-	panic("not implemented")
-}
-
-// ETCD use ip:port as unix socket address, however this address is invalid on windows.
-// We have to skip some of the test in such case.
-// https://github.com/etcd-io/etcd/blob/f0faa5501d936cd8c9f561bb9d1baca70eb67ab1/pkg/types/urls.go#L42
-func unixSocketAvailable() bool {
-	c, err := net.Listen("unix", "127.0.0.1:0")
-	if err == nil {
-		c.Close()
-		return true
-	}
-	return false
-}
-
 func TestInfo(t *testing.T) {
-	err := failpoint.Enable("github.com/pingcap/tidb/domain/infosync/FailPlacement", `return(true)`)
-	if err != nil {
-		t.Fatal(err)
-	}
-
 	if runtime.GOOS == "windows" {
 		t.Skip("integration.NewClusterV3 will create file contains a colon which is not allowed on Windows")
 	}
+
 	if !unixSocketAvailable() {
-		return
+		t.Skip("ETCD use ip:port as unix socket address, skip when it is unavailable.")
 	}
-	testleak.BeforeTest()
-	defer testleak.AfterTestT(t)()
-	ddlLease := 80 * time.Millisecond
+
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/domain/infosync/FailPlacement", `return(true)`))
+
 	s, err := mockstore.NewMockStore()
-	if err != nil {
-		t.Fatal(err)
-	}
-	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
-	defer clus.Terminate(t)
+	require.NoError(t, err)
+
+	cluster := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer cluster.Terminate(t)
+
 	mockStore := &mockEtcdBackend{
 		Storage: s,
-		pdAddrs: []string{clus.Members[0].GRPCAddr()}}
+		pdAddrs: []string{cluster.Members[0].GRPCAddr()}}
+	ddlLease := 80 * time.Millisecond
 	dom := NewDomain(mockStore, ddlLease, 0, 0, mockFactory)
 	defer func() {
 		dom.Close()
-		s.Close()
+		_ = s.Close()
```

Comment:
require assertion?

Reply:
Updated.

Let me think of the whole tests.